source/sn_coap_protocol.c

```diff
@@ -641,11 +639,21 @@ sn_coap_hdr_s *sn_coap_protocol_parse(struct coap_s *handle, sn_nsdl_addr_s *src
                 break;
             }
         }
-
+        /* Remove from the list if not an notification message.
+         * Initial notification message is needed for sending rest of the blocks (GET request).
+        */
+        bool remove_from_the_list = false;
         if (stored_blockwise_msg_temp_ptr) {
```

Comment:
This doable with one if
`if (stored_blockwise_msg_temp_ptr  &&
stored_blockwise_msg_temp_ptr->coap_msg_ptr && 
...
`